jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -128,6 +122,21 @@ def _on_viewer_data_changed(self, msg=None):
             # plugin, for example
             self.selected_spectrum = self.dc_items[0]
 
+    def _on_viewer_subsets_changed(self, msg):
+        """
+        Callback message for when a spectral subset is created, modified, or deleted.
+
+        Parameters
+        ----------
+        msg : `glue.core.Message`
+            The glue message passed to this callback method.
+        """
+        try:
+            self._spectral_subsets = self.app.get_subsets_from_viewer("spectrum-viewer",
+                                                                      subset_type="spectral")
+        except ValueError:
+            pass
+
```

Comment:
this shouldn't be needed any more as of #1156 (probably should have been removed as part of that PR)

Reply:
What do you mean? That the whole `def _on_viewer_subsets_changed` should be deleted?

Reply:
Removing this gives me a test failure, I think it might still be needed for the way the tests are calling things via the lower level API.

Reply:
So... what is the consensus?

Reply:
As discussed offline, I'll wait for @kecnry follow-up first before I review. Thanks!

Reply:
@pllim - block is now safely removed, you can continue your review!